indico/core/celery/__init__.py

```diff
@@ -45,7 +63,32 @@ def _import_modules(*args, **kwargs):
     signals.import_tasks.send()
 
 
+@beat_init.connect
+def _send_initial_heartbeat(*args, **kwargs):
+    heartbeat.delay(initial=True)
+
+
 @signals.menu.items.connect_via('admin-sidemenu')
 def _extend_admin_menu(sender, **kwargs):
     if session.user.is_admin:
         return SideMenuItem('celery', _("Tasks"), url_for('celery.index'), 20, icon='time')
+
+
+@template_hook('global-announcement', priority=-100)
+def _inject_announcement_header(**kwargs):
+    if not session.user or not session.user.is_admin or config.DISABLE_CELERY_CHECK:
+        return
+    last_ping = celery_settings.get('last_ping')
+    last_ping_version = celery_settings.get('last_ping_version')
+    down = not last_ping or (now_utc() - last_ping) > timedelta(hours=1)
+    mismatch = last_ping_version and last_ping_version != indico.__version__
+    if down or mismatch:
+        return render_template('celery/warning.html', down=down)
+
+
+@celery.periodic_task(name='heartbeat', run_every=crontab(minute='*/30'))
```

Comment:
maybe the ping interval should be  configurable as well?

Reply:
People can already override this via the config, but it's kind of pointless.